tests/test_service.py

```diff
@@ -0,0 +1,127 @@
+
+import pytest
+import logging
+
+from plumbum import ProcessExecutionError, local
+from plumbum.cmd import docker
+from plumbum.machines.local import LocalCommand
+
+logger = logging.getLogger()
+
+CONTAINER_NAME = "docksockprox_test"
+SOCKET_PROXY = "127.0.0.1:2375"
+
+
+def _start_proxy(
+    container_name=CONTAINER_NAME,
+    socket_proxy=SOCKET_PROXY,
+    extra_args=None
+):
+    logger.info(f"Starting {container_name} with args: {extra_args}...")
+    docker(
+        "run",
+        "-d",
+        "--name", container_name,
+        "--privileged",
+        "-v", "/var/run/docker.sock:/var/run/docker.sock",
+        "-p", f"{socket_proxy}:2375",
+        extra_args,
+        "tecnativa/docker-socket-proxy",
```

Comment:
It seems to me that you're gonna pull the latest build from docker hub if you do this.

You don't want to test that. You want to build an image and then test it. Please make sure you're doing that. Maybe it'll be easier if you add a pytest fixture that provides the locally built image name.